apps/src/templates/instructions/TopInstructions.story.jsx

```diff
@@ -0,0 +1,152 @@
+    pageConstants.ttsShortInstructionsUrl =
+      'https://tts.code.org/sharon22k/180/100/045539bb7fc9812eec4024867ac56d61/courseC_maze_programming8.mp3';
+
+    pageConstants.showNextHint = () => {};
+    instructionsConstants.noInstructionsWhenCollapsed = false;
+    store.dispatch(
+      setFeedback({
+        message:
+          'some simple, plaintext feedback, used to indicate that something went wrong'
+      })
+    );
+    store.dispatch(setHasAuthoredHints(true));
+    store.dispatch(
+      enqueueHints(
+        [
+          {
+            hintId: 'first',
+            markdown:
+              'this is the first hint. It has **some** _simple_ formatting'
+          },
+          {
+            hintId: 'second',
+            markdown:
+              'This is the second hint. It has an image.\n\n![](https://images.code.org/cab43107265a683a6216e18faab2353f-image-1452027548372.png)'
+          },
+          {
+            hintId: 'third',
+            markdown: 'This is the third hint. It has a Blockly block',
+            block: (
+              <xml>
+                <block type="maze_moveForward" />
+              </xml>
+            )
+          }
+        ],
+        []
+      )
+    );
+
+    pageConstants.showNextHint = () => {
+      store.dispatch(showNextHint());
+    };
+
+    pageConstants.smallStaticAvatar =
+      '/blockly/media/skins/bee/small_static_avatar.png';
+
+    pageConstants.failureAvatar = '/blockly/media/skins/bee/failure_avatar.png';
+  } else {
+    instructionsConstants.noInstructionsWhenCollapsed = true;
+    pageConstants.documentationUrl = 'https://docs.code.org/weblab/';
+    instructionsConstants.mapReference = '/docs/csd-1718/html_tags/index.html';
+    instructionsConstants.levelVideos = [
+      {
+        autoplay: true,
+        download:
+          'https://videos.code.org/levelbuilder/weblab_introtohtml-mp4.mp4',
+        enable_fallback: true,
+        key: 'csd_weblab_intro_2',
+        name: 'Intro to Web Lab - Part 2',
+        src:
+          'https://www.youtube-nocookie.com/embed/Hjl6gbg9kmk/?autoplay=1&enablejsapi=1&iv_load_policy=3&modestbranding=1&rel=0&showinfo=1&v=Hjl6gbg9kmk&wmode=transparent',
+        thumbnail: '/c/video_thumbnails/csd_weblab_intro_2.jpg'
+      }
+    ];
+  }
+
+  store.dispatch(setPageConstants(pageConstants));
+  store.dispatch(setInstructionsConstants(instructionsConstants));
+
+  return store;
+};
+
+const STORIES = {
+  'CSF Instructions': {
+    isCSF: true
+  },
+  'CSF Instructions Right-to-Left': {
+    rtl: true,
+    isCSF: true
+  },
+  'CSD/CSP Instructions': {
+    isCSF: false
+  },
+  'CSD/CSP Instructions Right-to-Left': {
+    rtl: true,
+    isCSF: false
+  }
+};
+
+export default storybook => {
+  const stories = storybook.storiesOf('TopInstructions', module);
+
+  Object.entries(STORIES).forEach(([name, options]) => {
```

Comment:
oh I like this set up - I don't think I've seen it before, but seems tidier than writing the stories out individually and passing different props combos

Reply:
Can't take credit for it - copied the setup that existed for InstructionsCSF